packages/in_app_purchase/in_app_purchase_android/lib/src/in_app_purchase_android_platform.dart

```diff
@@ -0,0 +1,327 @@
+            purchaseToken: purchaseParam.changeSubscriptionParam
+                ?.oldPurchaseDetails.verificationData.serverVerificationData,
+            prorationMode:
+                purchaseParam.changeSubscriptionParam?.prorationMode);
+    return billingResultWrapper.responseCode == BillingResponse.ok;
+  }
+
+  @override
+  Future<bool> buyConsumable(
+      {required PurchaseParam purchaseParam, bool autoConsume = true}) {
+    if (autoConsume) {
+      _productIdsToConsume.add(purchaseParam.productDetails.id);
+    }
+    return buyNonConsumable(purchaseParam: purchaseParam);
+  }
+
+  @override
+  Future<BillingResultWrapper> completePurchase(
+      PurchaseDetails purchase) async {
+    assert(
+      purchase is GooglePlayPurchaseDetails,
+      'On Android, the `purchase` should always be of type `GooglePlayPurchaseDetails`.',
+    );
+
+    GooglePlayPurchaseDetails googlePurchase =
+        purchase as GooglePlayPurchaseDetails;
+
+    if (googlePurchase.billingClientPurchase.isAcknowledged) {
+      return BillingResultWrapper(responseCode: BillingResponse.ok);
+    }
+
+    if (googlePurchase.verificationData == null) {
+      throw ArgumentError(
+          'completePurchase unsuccessful. The `purchase.verificationData` is not valid');
+    }
+
+    return await billingClient
+        .acknowledgePurchase(purchase.verificationData.serverVerificationData);
+  }
+
+  /// Mark that the user has consumed a product.
+  ///
+  /// You are responsible for consuming all consumable purchases once they are
+  /// delivered. The user won't be able to buy the same product again until the
+  /// purchase of the product is consumed.
+  Future<BillingResultWrapper> consumePurchase(PurchaseDetails purchase) {
+    if (purchase.verificationData == null) {
+      throw ArgumentError(
+          'consumePurchase unsuccessful. The `purchase.verificationData` is not valid');
+    }
+    return billingClient
+        .consumeAsync(purchase.verificationData.serverVerificationData);
+  }
+
+  @override
+  Future<void> restorePurchases({
+    String? applicationUserName,
+  }) async {
+    List<PurchasesResultWrapper> responses;
+
+    responses = await Future.wait([
+      billingClient.queryPurchases(SkuType.inapp),
+      billingClient.queryPurchases(SkuType.subs)
+    ]);
+
+    Set errorCodeSet = responses
+        .where((PurchasesResultWrapper response) =>
+            response.responseCode != BillingResponse.ok)
+        .map((PurchasesResultWrapper response) =>
+            response.responseCode.toString())
+        .toSet();
+
+    String errorMessage =
+        errorCodeSet.isNotEmpty ? errorCodeSet.join(', ') : '';
+
+    List<PurchaseDetails> pastPurchases =
+        responses.expand((PurchasesResultWrapper response) {
+      return response.purchasesList;
+    }).map((PurchaseWrapper purchaseWrapper) {
+      return GooglePlayPurchaseDetails.fromPurchase(purchaseWrapper);
```

Comment:
Do we need to set the status to be "restored"?

Reply:
Yes is should. Updated thanks.